tests/evolution/test_checkpointing.py

```
@@ -0,0 +1,366 @@
+"""
+Tests for evolution checkpointing and recovery functionality.
+
+This module tests the ability to save and restore evolution state,
+enabling recovery from failures and resumption of long-running evolutions.
+"""
+
+import json
+import os
+import tempfile
+from pathlib import Path
+from unittest.mock import AsyncMock, MagicMock, patch
```

Ruff (0.12.2)

`unittest.mock.patch` imported but unused

Remove unused import: `unittest.mock.patch`

(F401)